File: backend/services/CourseService/CourseService.Shared/Paging/PagedResult.cs
namespace CourseService.Shared.Paging
{
    public class PagedResult<T>
    {
        public int Skip { get; private set; }
        public int Limit { get; private set; }
        public long Total { get; private set; }
        public List<T> Items { get; private set; }

        public PagedResult(List<T> items, int skip, int limit, long total)
        {
            Items = items;
            Skip = skip;
            Limit = limit;
            Total = total;
        }
    }
}

File: frontend/vicourses_mobile_app/lib/utils/paged_result.dart
class PagedResult<T> {
  PagedResult({
    required this.items,
    this.total = 0,
    this.skip = 0,
    this.limit = 0,
    this.end = false,
  });

  final List<T> items;
  final int total;
  final int skip;
  final int limit;
  final bool end;
}
